### PR TITLE
Google Cloud Monitor: Prom query editor

### LINKED ISSUE #70962: Add PromQL to Google Cloud Monitoring
Use the monocao editor package to create something like the [QueryField in Prometheus](https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/prometheus/components/PromQueryField.tsx). 

Alternatively, we use a text input for now that allows rudimentary `PromQL` but no additional features like syntax highlighting or tab completion but allows us to ship quicker and iterate down the road. 

Maybe https://github.com/prometheus-community/monaco-promql helps here?

### PR DESCRIPTION
**Pull request by bossinc**
Prometheus queries in gcm! https://github.com/grafana/grafana/issues/70962

Currently, there is just a textbox for the expression editor. I can look into using https://github.com/prometheus-community/monaco-promql

### CODE REVIEW

File: pkg/tsdb/cloud-monitoring/promql_query.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+		VectorWideSeries: false,
+		Dataplane:        false,
+	})
+
+	// Add frame to attach metadata
+	if len(r.Frames) == 0 {
+		r.Frames = append(r.Frames, data.NewFrame(""))
+	}
+
+	*queryRes = r
```
Comment (aangelisc) [inline pkg/tsdb/cloud-monitoring/promql_query.go:91]: Interesting way of doing this, why not set `queryRes.Frames = r.Frames`?

**Review by lwandz13 — APPROVED** [review]

File: public/app/plugins/datasource/cloud-monitoring/components/PromQLEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            css`
+              flex-wrap: nowrap;
+            `
+          )}
+          aria-label="Step field"
+        >
+          <InlineFormLabel
+            width={6}
+            tooltip={
+              'Time units and built-in variables can be used here, for example: $__interval, $__rate_interval, 5s, 1m, 3h, 1d, 1y (Default if no unit is specified: s)'
```
Comment (aangelisc) [inline public/app/plugins/datasource/cloud-monitoring/components/PromQLEditor.tsx:76]:
```suggestion
              'Time units and built-in variables can be used here, for example: $__interval, $__rate_interval, 5s, 1m, 3h, 1d, 1y (Default if no unit is specified: 10s)'
```

File: public/app/plugins/datasource/cloud-monitoring/dataquery.cue [hunk 3]
```diff
@@ -23,7 +23,7 @@ composableKinds: DataQuery: {
 
 	lineage: {
 		schemas: [{
-			version: [0, 0]
+			version: [0, 0] //do i update this?
```
Comment (aangelisc) [inline public/app/plugins/datasource/cloud-monitoring/dataquery.cue:26]: I'm not sure, @sdboyer should this be automatically or manually updated?

**Review by aangelisc — APPROVED** [review]
Great stuff @bossinc! I've left some comments that you can address/ignore 😊 I like this and think it's a valuable change!

File: pkg/tsdb/cloud-monitoring/promql_query.go (hunk 1 above)
Comment (bossinc) [reply pkg/tsdb/cloud-monitoring/promql_query.go:91]: If we just get the frames, then we won't get the metadata from the `ReadPrometheusStyleResult`
As a side note or potential refactor, I think how its strange how `parseResponse` isn't returning a `DataResponse`. I'm not sure why `DataResponse` is part of the struct

**Review by asimpson — APPROVED** [review]
Nice work on this, excited to have it go live! :rocket:

**Pull request closed without merging**